src/Fantomas/CodeFormatterImpl.fs

```diff
@@ -359,29 +354,26 @@ let isValidFSharpCode (checker: FSharpChecker) (parsingOptions: FSharpParsingOpt
             let! ast = parse checker parsingOptions formatContext
 
             let isValid =
-                ast |> Array.map fst |> Array.forall isValidAST
+                ast
+                |> Array.map (fun (a, _, _) -> a)
```

Comment:
Obvious allocation reduction: `ast |> Array.forall (fun (a, _, _) -> isValidAST a)`. But I imagine this is unlikely ever to matter.

Reply:
Good spot, never hesitate to mention things like this!